packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3458,3 +3462,61 @@ const waitForSeq = async (
 			deltaManager.on("op", handleOp);
 		}
 	});
+
+/**
+ * Mixin class that adds await for DataObject to finish initialization before we proceed to summary.
+ * @param handler - handler that returns info about blob to be added to summary.
+ * Or undefined not to add anything to summary.
+ * @param Base - base class, inherits from FluidDataStoreRuntime
+ */
+export const mixinSummaryHandler = (
+	handler: (
+		runtime: ContainerRuntime,
+	) => Promise<{ path: string[]; content: string } | undefined>,
+	Base: typeof ContainerRuntime = ContainerRuntime,
+) =>
+	class RuntimeWithSummarizerHandler extends Base {
+		async summarize(options) {
+			const result = await super.summarize(options);
+			const content = await handler(this);
+			if (content !== undefined) {
+				addBlobToSummaryViaPath(
+					result.summary,
+					result.stats,
+					content.path,
+					content.content,
+				);
+			}
+			return result;
+		}
+	} as typeof ContainerRuntime;
+
+// Helper API to load extra blobs in snapshot
+// This API only works for blobs that are saved in storage, i.e. this API would only work for load from storage scenarios
+export async function loadRuntimeBlob(params: IContainerRuntimeParams, path: string[]) {
+	const context = params.context;
+	const pendingRuntimeState = context.pendingLocalState as IPendingRuntimeState | undefined;
+	let baseSnapshot = context.baseSnapshot;
+	const storage = context.storage;
+
+	// Load custom application blob
+	// This state is only available for real snapshots, we do not write that extra state when serializing local (uncommited) changes.
+	let extraBlob: string | undefined;
+	if (pendingRuntimeState === undefined) {
+		const blobName = path.pop();
+		for (const name of path) {
+			if (baseSnapshot) {
+				baseSnapshot = baseSnapshot?.trees[name];
+			}
+		}
+
+		if (baseSnapshot && blobName) {
+			// We got the blob!
+			assert(storage !== undefined, "attached state should have storage");
+			extraBlob = await readAndParse<string>(storage, baseSnapshot?.blobs[blobName]);
```

Comment:
should we just return here when we find it? And return 'undefined` after the topmost 'if'? We'd get rid of the `extraBlob` variable.